discovery-provider/src/tasks/index_network_peers.py

```diff
@@ -0,0 +1,174 @@
+        content_node_service_type,
+        sp_id
+    ).call()
+    return cn_endpoint_info
+
+# Query the L1 set of audius protocol contracts and retrieve a list of peer endpoints
+def retrieve_peers_from_eth_contracts(self):
+    shared_config = update_network_peers.shared_config
+    eth_web3 = update_network_peers.eth_web3
+    eth_registry_address = update_network_peers.eth_web3.toChecksumAddress(
+        shared_config["eth_contracts"]["registry"]
+    )
+    eth_registry_instance = eth_web3.eth.contract(
+        address=eth_registry_address, abi=eth_abi_values["Registry"]["abi"]
+    )
+    sp_factory_address = eth_registry_instance.functions.getContract(
+        sp_factory_registry_key
+    ).call()
+    sp_factory_inst = eth_web3.eth.contract(
+        address=sp_factory_address, abi=eth_abi_values["ServiceProviderFactory"]["abi"]
+    )
+    total_cn_type_providers = sp_factory_inst.functions.getTotalServiceTypeProviders(content_node_service_type).call()
+    ids_list = list(range(1, total_cn_type_providers + 1))
+    eth_cn_endpoints = {}
+    # Given the total number of nodes in the network we can now fetch node info in parallel
+    with concurrent.futures.ThreadPoolExecutor(max_workers=5) as executor:
+        fetch_cnode_futures = {executor.submit(fetch_cnode_info, i, sp_factory_inst): i for i in ids_list}
+        for future in concurrent.futures.as_completed(fetch_cnode_futures):
+            single_cnode_fetch_op = fetch_cnode_futures[future]
+            try:
+                cn_endpoint_info = future.result()
+                # 1 index entry in the returned array from ServiceProviderFactory
+                eth_cn_endpoints[cn_endpoint_info[1]] = True
+            except Exception as exc:
+                logger.error(f"index_network_peers.py | fetch_cnode_futures {single_cnode_fetch_op} generated {exc}")
+    # Return dictionary with key = endpoint, formatted as { endpoint: True }
+    return eth_cn_endpoints
+
+# Determine the known set of distinct peers currently within a user replica set
+# This function differs from the above as we are not interacting with eth-contracts,
+#   instead we are pulling local db state and retrieving the relevant information
+def retrieve_peers_from_db(self):
+    db = update_network_peers.db
+    cnode_endpoints = {}
+    with db.scoped_session() as session:
+        db_cnode_endpts = (
+            session.query(
+                User.creator_node_endpoint).filter(
+                    User.creator_node_endpoint != None, User.is_current == True
+                ).distinct()
+        )
+        # Generate dictionary of unique creator node endpoints
+        for entry in db_cnode_endpts:
+            for cnode_user_set in entry:
+                cnode_entries = cnode_user_set.split(',')
+                for cnode_url in cnode_entries:
+                    if cnode_url == "''":
+                        continue
+                    cnode_endpoints[cnode_url] = True
```

Comment:
nit: could use a `Set` for this to avoid having to set the value to True

Reply:
I think sets in python are immutable